Sources/SafeDICore/Models/TypeDescription.swift

```diff
@@ -244,6 +244,17 @@ public enum TypeDescription: Codable, Hashable, Comparable, Sendable {
         }
     }
 
+    var strippingGenerics: TypeDescription {
+        switch self {
+        case let .simple(name, _):
+            .simple(name: name, generics: [])
+        case let .nested(name, parentType, _):
+            .nested(name: name, parentType: parentType, generics: [])
+        case .array, .attributed, .any, .closure, .composition, .dictionary, .implicitlyUnwrappedOptional, .metatype, .optional, .some, .tuple, .unknown, .void:
```

Comment:
it's possible to strip generics out of some of these types (like optionals)... but doing so doesn't really mean anything to SafeDI today. Keeping this on the simple side until we need it.